sdk/metricsadvisor/azure-ai-metricsadvisor-perf/src/main/java/com/azure/ai/metricsadvisor/perf/core/ServiceTest.java

```diff
@@ -0,0 +1,96 @@
+     * Creates an instance of performance test.
+     *
+     * @param options the options configured for the test.
+     */
+    public ServiceTest(TOptions options) {
+        super(options);
+
+        final String endpoint = Configuration.getGlobalConfiguration().get("METRICS_ADVISOR_ENDPOINT");
+        if (CoreUtils.isNullOrEmpty(endpoint)) {
+            throw logger.logExceptionAsError(
+                new RuntimeException(String.format(CONFIGURATION_ERROR, "METRICS_ADVISOR_ENDPOINT")));
+        }
+
+        final String subscriptionKey = Configuration.getGlobalConfiguration().get("METRICS_ADVISOR_SUBSCRIPTION_KEY");
+        if (CoreUtils.isNullOrEmpty(subscriptionKey)) {
+            throw logger.logExceptionAsError(
+                new RuntimeException(String.format(CONFIGURATION_ERROR, "METRICS_ADVISOR_SUBSCRIPTION_KEY")));
+        }
+
+        final String apiKey = Configuration.getGlobalConfiguration().get("METRICS_ADVISOR_API_KEY");
+        if (CoreUtils.isNullOrEmpty(apiKey)) {
+            throw logger.logExceptionAsError(
+                new RuntimeException(String.format(CONFIGURATION_ERROR, "METRICS_ADVISOR_API_KEY")));
+        }
+
+        MetricsAdvisorClientBuilder builder = new MetricsAdvisorClientBuilder()
+            .endpoint(endpoint)
+            .credential(new MetricsAdvisorKeyCredential(subscriptionKey, apiKey));
+
+        this.metricsAdvisorClient = builder.buildClient();
+        this.metricsAdvisorAsyncClient = builder.buildAsyncClient();
+
+        this.alertConfigId = Configuration.getGlobalConfiguration().get("METRICS_ADVISOR_ALERT_CONFIG_ID");
+        if (CoreUtils.isNullOrEmpty(alertConfigId)) {
+            throw logger.logExceptionAsError(
+                new RuntimeException(String.format(CONFIGURATION_ERROR, "METRICS_ADVISOR_ALERT_CONFIG_ID")));
+        }
+
+        this.alertId = Configuration.getGlobalConfiguration().get("METRICS_ADVISOR_ALERT_ID");
+        if (CoreUtils.isNullOrEmpty(alertId)) {
+            throw logger.logExceptionAsError(
+                new RuntimeException(String.format(CONFIGURATION_ERROR, "METRICS_ADVISOR_ALERT_ID")));
+        }
+
+        this.detectionConfigId
+            = Configuration.getGlobalConfiguration().get("METRICS_ADVISOR_DETECTION_CONFIG_ID");
+        if (CoreUtils.isNullOrEmpty(detectionConfigId)) {
+            throw logger.logExceptionAsError(
+                new RuntimeException(String.format(CONFIGURATION_ERROR, "METRICS_ADVISOR_DETECTION_CONFIG_ID")));
+        }
+
+        this.incidentId = Configuration.getGlobalConfiguration().get("METRICS_ADVISOR_INCIDENT_ID");
+        if (CoreUtils.isNullOrEmpty(incidentId)) {
+            throw logger.logExceptionAsError(
+                new RuntimeException(String.format(CONFIGURATION_ERROR, "METRICS_ADVISOR_INCIDENT_ID")));
+        }
+
+        this.maxListElements = Configuration.getGlobalConfiguration()
+            .get("METRICS_ADVISOR_MAX_LIST_ELEMENTS", DEFAULT_MAX_LIST_ELEMENTS);
```

Comment:
instead of using env vars, to pass in these properties for test scenarios, use the options bag.
Create your custom options bag which extends from PerfStressOptions and pass those in your Perf Test classes that extend the Service Test Class. 
OPtions bag will allow you to consume options from command line instead of env var and will align the usage across languages.